SmartReceipts/Modules/Backup/BackupInteractor.swift

```
@@ -29,10 +29,81 @@ class BackupInteractor: Interactor {
         return purchaseService.hasValidSubscriptionValue()
     }
     
+    func downloadZip(_ backup: RemoteBackupMetadata) {
+        weak var hud = PendingHUDView.showFullScreen()
+        backupManager.downloadAllData(remoteBackupMetadata: backup)
+            .map({ result -> URL? in
+                if !result.database.open() { return nil }
```

Maybe have this throw an error instead of `return nil`?

---

@wbaumann good point, i'll change this